src/robot/utils/connectioncache.py

```diff
@@ -100,13 +100,28 @@ def get_connection(self, alias_or_index=None):
                 self.current.raise_error()
             return self.current
         try:
-            index = self.resolve_alias_or_index(alias_or_index)
+            index = self._resolve_connection(alias_or_index)
         except ValueError as err:
             raise RuntimeError(err.args[0])
         return self._connections[index-1]
 
     __getitem__ = get_connection
 
+    def get_connection_index(self,popen=None):
+        """Returns index value of process(popen) object in connections list.
+
+            If ``popen`` is ``None``, returns the current connection
+            if it is active, or raises an error if it is not.
+        """
+        if popen is None:
+            if not self:
+                self.current.raise_error()
+            return self._connections.index(self.current)+1
+        try:
+            return self._connections.index(popen)+1
+        except ValueError:
+            raise ValueError(f"Provided input: '{popen}' is not a process object (popen).")
+
```

Comment:
This code seems to be only used by the keyword that I believe isn't needed. Thus this code can be removed as well. If it is preserved, it should be made independent also on terminology level from Popen.

Reply:
removed index getter method